dbt/adapters/base/impl.py

```diff
@@ -1787,6 +1787,29 @@ def _get_adapter_specific_run_info(cls, config) -> Dict[str, Any]:
         """
         return {}
 
+    @available.parse_none
+    @classmethod
+    def get_hard_deletes_behavior(cls, config):
+        """Check the hard_deletes config enum, and the legacy invalidate_hard_deletes
+        config flag in order to determine which behavior should be used for deleted
+        records in a snapshot. The default is to ignore them."""
+        invalidate_hard_deletes = config.get("invalidate_hard_deletes", None)
+        hard_deletes = config.get("hard_deletes", None)
+
+        if invalidate_hard_deletes is not None and hard_deletes is not None:
```

Comment:
Do we want to raise an error? Or do we just want to respect the legacy setting (`invalidate_hard_deletes`) for backwards compatibility? I guess I'm also assuming `hard_deletes` ultimately replaces `invalidate_hard_deletes`.

Reply:
I think raising an error is the way to go here, since the most likely scenario is that the user has made a mistake. The only potential impact on backwards compatibility would be if they had set a value of hard_deletes already, which seems highly unlikely.

Reply:
After discussion we decided that setting both invalid_hard_deletes and hard_deletes should be a validation error. Do you have a strong opinion on this @graciegoheen ?